python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,581 @@
+        psf : `lsst.afw.detection.Psf`
+            PSF model for this exposure.
+        background : `lsst.afw.math.BackgroundList`
+            Total background that had been fit to the exposure so far;
+            modified in place to include background fit when detecting sources.
+        ap_corr : `lsst.afw.image.ApCorrMap`
+            Aperture Correction model for this exposure.
+        photo_calib : `lsst.afw.image.PhotoCalib`
+            Photometric calibration model for this exposure.
+        wcs : `lsst.afw.geom.SkyWcs`
+            World Coordinate System model for this exposure.
+        summary_stats : `lsst.afw.image.ExposureSummaryStats`
+            Summary statistics measured on this exposure.
+        calib_sources : `astropy.table.Table`
+            Per-visit catalog of measurements to get 'calib_*' flags from.
+        id_generator : `lsst.meas.base.IdGenerator`, optional
+            Object that generates source IDs and provides random seeds.
+        result : `lsst.pipe.base.Struct`, optional
+            Result struct that is modified to allow saving of partial outputs
+            for some failure conditions. If the task completes successfully,
+            this is also returned.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results as a struct with attributes:
+
+            ``exposure``
+                Calibrated exposure, with pixels in nJy units.
+                (`lsst.afw.image.Exposure`)
+            ``sources``
+                Sources that were measured on the exposure, with calibrated
+                fluxes and magnitudes. (`astropy.table.Table`)
+            ``sources_footprints``
+                Footprints of sources that were measured on the exposure.
+                (`lsst.afw.table.SourceCatalog`)
+            ``background``
+                Total background that was fit to, and subtracted from the
+                exposure when detecting ``sources``, in the same nJy units as
+                ``exposure``. (`lsst.afw.math.BackgroundList`)
+        """
+        if result is None:
+            result = pipeBase.Struct()
+        if id_generator is None:
+            id_generator = lsst.meas.base.IdGenerator()
+
+        result.exposure = self.snap_combine.run(exposures).exposure
+
+        # Calibrate the image, so it's on the same units as the background.
+        result.exposure.maskedImage = initial_photo_calib.calibrateImage(result.exposure.maskedImage)
+        result.exposure.maskedImage -= background.getImage()
+        # Uncalibrate so that we do the measurements in instFlux, because
+        # we don't have a way to identify measurements as being in nJy (TODO!).
+        result.exposure.maskedImage /= initial_photo_calib.getCalibrationMean()
+
+        result.exposure.setPsf(psf)
+        result.exposure.setApCorrMap(ap_corr)
+        result.exposure.setWcs(wcs)
+        result.exposure.info.setSummaryStats(summary_stats)
```

Comment:
So we reuse the summary stats and don't recompute based on the final measurements?

Reply:
Good question. I suppose that is worth recalculating at the end instead. Then we shouldn't have to pass the summary stats in, which means not having to pick the correct entry out of the visit list.

Reply:
Ah, scratch the "not having to pick" part: I still need the visit summary for the psf, wcs, etc.